packages/auto-id/src/auto-id.ts

```diff
@@ -87,15 +87,6 @@ function useId(
 function useId(): string | undefined;
 
 function useId(providedId?: number | string | undefined | null) {
-	// TODO: Remove error flag when updating internal deps to React 18. None of
-	// our tricks will play well with concurrent rendering anyway.
-	// @ts-expect-error
-	if (typeof React.useId === "function") {
-		// @ts-expect-error
-		let id = React.useId(providedId);
-		return providedId != null ? providedId : id;
-	}
```

Comment:
This is what is causing the problem. I suggest to simply remove this once only React >=18 is supported.

Reply:
Is there no way to rewrite this to keep your build process from showing a warning? I think that would be preferable.